### ISSUE TITLE
Support for $ projection

### ISSUE THREAD

**Issue by Maxeh** [bwaldvogel/mongo-java-server#104]
Hello, first of all thanks for the great work :)

We are using a custom query in our project with [projection](https://docs.mongodb.com/manual/reference/operator/projection/positional/), but it seems that mongo-java-server does not support such kind of projection. The query looks for objects which have a specific nested object in an array and the query returns the object with only the matching nested object in the array. It looks like this:

`@Query(value = "{ 'states': { $elemMatch: { 'state': { $eq: \"INSTALLED\" }, 'internalID': { $eq: ?0 } } } }", fields = "{ 'states.$': 1, 'serialnumber': 1 }")`
`List<MyObject> findWithCustomQuery(String internalID);`

So if we have a object like this:
```
object1:
|---serialnumber: 1234
|---states:
     |---[0]
           |---state: INSTALLED
           |---internalID: abc123
     |---[1]
           |---state: UPDATE
           |---internalID: abc456
object2:
|---serialnumber: 5678
|---states:
     |---[0]
           |---state: UPDATE
           |---internalID: abc123
     |---[1]
           |---state: UPDATE
           |---internalID: abc456
```

and the query is done with internalID abc123 only the first object containing an array with the matching object should be returned. This _works in mongo 3.6_, but not with mongo-java-server. If I use mongo-java-server all fields of the array are set to null and all objects in the array are returned. So the expected result would be:

```
object1:
|---serialnumber: 1234
|---states:
     |---[0]
           |---state: INSTALLED
           |---internalID: abc123
```
But we actually get:
```
object1:
|---serialnumber: 1234
|---states:
     |---[0]
           |---state: null
           |---internalID: null
     |---[1]
           |---state: null
           |---internalID: null

```
Any help is appreciated!

**Comment by markx3** [comment]
I'm experiencing a similar issue using `$avg` on projections too.

This closed issue might be related: #68.

In my case, I was trying to do something in the likes:

```java
...
project().and(AccumulatorOperators.Avg.avgOf("foo.bar")).as("foobar"))
         .and(...)
...
```

Which throws `Unrecognized expression '$avg'`. Could these be related?

Appreciate any input!

**Comment by bwaldvogel** [comment]
@Maxeh: This should be fixed on master thanks to #108. Could you re-test?

**Comment by bwaldvogel** [comment]
> @markx3: In my case, I was trying to do something in the likes: 
> ```java
> project().and(AccumulatorOperators.Avg.avgOf("foo.bar")).as("foobar"))
>          .and(...)
> ```
> Which throws `Unrecognized expression '$avg'`. Could these be related?

This is actually a new issue and neither related to #104 nor #68.
Anyway, I’ve [improved the support](https://github.com/bwaldvogel/mongo-java-server/commit/eb3810485feea6ad7664139b906b9bb5557f7609) for `$min`, `$max` and `$avg` expressions as you outlined in your comment. Can you re-test and open a new issue if it still doesn’t work for you?
Thanks!

**Comment by Maxeh** [comment]
Thanks for the update @bwaldvogel :)

What is the best way to test the new master version locally in maven? I tried jitpack.io, but the tool is not able to build the project:

```
* What went wrong:
A problem occurred evaluating root project 'build'.
> Project with path ':mongo-java-server-core' could not be found in root project 'build'.
```

**Comment by bwaldvogel** [comment]
You could try to build a "fat" JAR as explained in https://github.com/bwaldvogel/mongo-java-server#building-a-fat-jar-that-contains-all-dependencies

**Comment by Maxeh** [comment]
@bwaldvogel I have created a fat jar file and installed it to my local maven repo and it worked. 

The bug seems to be fixed and the correct data is loaded. When do you think will the new version be available in the maven repository?

**Comment by bwaldvogel** [comment]
I’ve just released version `1.23.0`.
